### PR TITLE
Feat: add sequence enumerate op

### PR DESCRIPTION
**Pull request by chenwhql**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_sequence_enumerate_op.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        self.in_seq = np.random.randint(0, 10, (30, 1)).astype("int32")
+        self.lod = [[9, 4, 11, 6]]
+        self.win_size = 2
+        self.pad_value = 0
+        out_seq = sequence_enumerate(self.in_seq, self.lod, self.win_size,
+                                     self.pad_value)
+        self.out_seq = np.array(out_seq).astype("int32")
+
+
+class TesSequenceEnumerateOpInt64(TestSequenceEnumerateOp):
```
Comment (kuke) [inline python/paddle/fluid/tests/unittests/test_sequence_enumerate_op.py:60]: Please add a test case for large window size, e.g., 3 or 5.

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5649,6 +5650,52 @@ def flatten(x, axis=1, name=None):
     return out
 
 
+def sequence_enumerate(input, win_size, pad_value, name=None):
+    """
+    Generate a new LoDTensor
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: Generate a new sequence for the input index sequence, which enumerates  all the sub-sequences with length `win_size` of the input. The enumerated sequence has the same 1st dimension with Variable `input`,  and the 2nd dimension is `win_size`, padded by `pad_value` if necessary in generation.

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -5649,6 +5650,52 @@ def flatten(x, axis=1, name=None):
...
+      Attrs:
+        win_size = 2
+        pad_value = 0
+      Output:
+        Out.lod = [[0, 3, 5]]
+        Out.data = [[1, 2], [2, 3], [3, 0], [4, 5], [5, 0]]
+        Out.dims = [5, 2]
+
+    Args:
+        input (Variable): The input variable which is a LoDTensor
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: The input Variable which usually is an index sequence.

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -5649,6 +5650,52 @@ def flatten(x, axis=1, name=None):
     return out
 
 
+def sequence_enumerate(input, win_size, pad_value, name=None):
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `pad_value=0`

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -5649,6 +5650,52 @@ def flatten(x, axis=1, name=None):
...
+        win_size = 2
+        pad_value = 0
+      Output:
+        Out.lod = [[0, 3, 5]]
+        Out.data = [[1, 2], [2, 3], [3, 0], [4, 5], [5, 0]]
+        Out.dims = [5, 2]
+
+    Args:
+        input (Variable): The input variable which is a LoDTensor
+        win_size (int): The enumerate sequence window size.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: -> `win_size (int): The window size for enumerating all sub-sequences.`

File: python/paddle/fluid/layers/nn.py [hunk 6]
```diff
@@ -5649,6 +5650,52 @@ def flatten(x, axis=1, name=None):
...
+        pad_value (int): The enumerate sequence padding value.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `pad_value (int): The padding value, default 0.`

File: paddle/fluid/operators/sequence_enumerate_op.cc [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        .AddCustomChecker([](const int& win_size) {
+          PADDLE_ENFORCE(win_size >= 2,
+                         "The window size should be greater than 2.");
+        });
+    AddAttr<int>("pad_value", "(int) The enumerate sequence padding value.")
+        .SetDefault(0);
+    AddComment(R"DOC(
+Sequence Enumerate Operator.
+
+Sequence enumerate operator generate a new LoDTensor
```
Comment (kuke) [inline paddle/fluid/operators/sequence_enumerate_op.cc]: Please refine this doc by following the comment under python API.

File: paddle/fluid/operators/sequence_enumerate_op.cc [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+class SequenceEnumerateOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X",
+             "(2-D LoDTensor with the 2nd dimension equal to 1) "
+             "Input LoDTensor of SequenceEnumerate operator.");
+    AddOutput("Out",
+              "(2-D LoDTensor with the 2nd dimension equal to 1) "
```
Comment (kuke) [inline paddle/fluid/operators/sequence_enumerate_op.cc]: Wrong dimension description

File: paddle/fluid/operators/sequence_enumerate_op.cc [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+              "Output LoDTensor of SequenceEnumerate operator.");
+    AddAttr<int>("win_size", "(int) The enumerate sequence window size.")
+        .AddCustomChecker([](const int& win_size) {
+          PADDLE_ENFORCE(win_size >= 2,
+                         "The window size should be greater than 2.");
```
Comment (kuke) [inline paddle/fluid/operators/sequence_enumerate_op.cc]: `greater` -> `not less`

File: python/paddle/fluid/tests/unittests/test_sequence_enumerate_op.py (hunk 1 above)
Comment (chenwhql) [reply python/paddle/fluid/tests/unittests/test_sequence_enumerate_op.py:60]: Done

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (chenwhql) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (chenwhql) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 4 above)
Comment (chenwhql) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 5 above)
Comment (chenwhql) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 6 above)
Comment (chenwhql) [reply python/paddle/fluid/layers/nn.py]: Done

File: paddle/fluid/operators/sequence_enumerate_op.cc (hunk 7 above)
Comment (chenwhql) [reply paddle/fluid/operators/sequence_enumerate_op.cc]: Done

File: paddle/fluid/operators/sequence_enumerate_op.cc (hunk 8 above)
Comment (chenwhql) [reply paddle/fluid/operators/sequence_enumerate_op.cc]: Done

File: paddle/fluid/operators/sequence_enumerate_op.cc (hunk 9 above)
Comment (chenwhql) [reply paddle/fluid/operators/sequence_enumerate_op.cc]: Done

File: paddle/fluid/API.spec [hunk 10]
```diff
@@ -164,6 +164,7 @@ paddle.fluid.layers.crop ArgSpec(args=['x', 'shape', 'offsets', 'name'], varargs
 paddle.fluid.layers.rank_loss ArgSpec(args=['label', 'left', 'right', 'name'], varargs=None, keywords=None, defaults=(None,))
 paddle.fluid.layers.prelu ArgSpec(args=['x', 'mode', 'param_attr', 'name'], varargs=None, keywords=None, defaults=(None, None))
 paddle.fluid.layers.flatten ArgSpec(args=['x', 'axis', 'name'], varargs=None, keywords=None, defaults=(1, None))
+paddle.fluid.layers.sequence_enumerate ArgSpec(args=['input', 'win_size', 'pad_value', 'name'], varargs=None, keywords=None, defaults=(None,))
```
Comment (kuke) [inline paddle/fluid/API.spec]: Need to regenerate API spec after changing.

File: python/paddle/fluid/layers/nn.py [hunk 11]
```diff
@@ -5649,6 +5650,51 @@ def flatten(x, axis=1, name=None):
     return out
 
 
+def sequence_enumerate(input, win_size, pad_value=0, name=None):
+    """
+    Generate a new sequence for the input index sequence, which enumerates all the
+    sub-sequences with length win_size of the input.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: win_size -> `win_size` 
pad_value -> `pad_value`
在doc中出现这些输入变量时一定要加引用

File: paddle/fluid/API.spec (hunk 10 above)
Comment (chenwhql) [reply paddle/fluid/API.spec]: Done

File: python/paddle/fluid/layers/nn.py (hunk 11 above)
Comment (chenwhql) [reply python/paddle/fluid/layers/nn.py]: Done

**Review by kuke — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]
approve for API.

**Review by reyoung — APPROVED** [review]

**Pull request merged**